src/rust/engine/src/scheduler.rs

```diff
@@ -45,6 +45,16 @@ impl Scheduler {
     self.graph.visualize(&self.roots, path, &self.tasks.externs)
   }
 
+  pub fn trace(&self, path: &Path) -> io::Result<()> {
+    for root in &self.roots {
+      let result = self.graph.trace(&root, path, &self.tasks.externs);
```

Comment:
It should be equivalent to wrap this line in `try!()`... equivalent the new `?` operator: https://blog.rust-lang.org/2016/11/10/Rust-1.13.html